### PR TITLE
Backport/optional log fill attr from profiles [6740]

### PR DESCRIPTION
**Pull request by EduPonz**
This PR is linked with [ros2/rmw_fastrtps 355 PR](https://github.com/ros2/rmw_fastrtps/pull/335). It has been done so that the RMW layer does not show log errors when a profile is not found.
* Add an optional parameter `log_error` for only logging the error when user specifies on member functions:
    * `XMLProfileManager::fillParticipantAttributes`
    * `XMLProfileManager::fillPublisherAttributes`
    * `XMLProfileManager::fillSubscriberAttributes`
* Style compliance on:
    * XMLProfileManager.h
    * XMLProfileManager.cpp

### CODE REVIEW

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 1]
```diff
@@ -117,11 +121,13 @@ class XMLProfileManager
...
         * @param profile_name Name for the profile to be used to fill the structure.
         * @param atts Structure to be filled.
+        * @param log_error Flag to log an error if the profile_name is not found.
         * @return XMLP_ret::XML_OK on success, XMLP_ret::XML_ERROR in other case.
         */
         RTPS_DllAPI static XMLP_ret fillSubscriberAttributes(
                 const std::string &profile_name,
-                SubscriberAttributes &atts);
+                SubscriberAttributes &atts,
+                bool log_error = false);
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply include/fastrtps/xmlparser/XMLProfileManager.h]: done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/786/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/663/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/781/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/236/)

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 2]
```diff
@@ -95,12 +95,12 @@ class XMLProfileManager
      * @param profile_name Name for the profile to be used to fill the structure.
      * @param atts Structure to be filled.
      * @param log_error Flag to log an error if the profile_name is not found.
-     * @return XMLP_ret::XML_OK on success, XMLP_ret::XML_ERROR in other case.
+     * @return XMLP_ret::XML_OK on success, XMLP_ret::XML_ERROR in other case. Defaults true.
```
Comment (MiguelCompany) [inline include/fastrtps/xmlparser/XMLProfileManager.h]: This should go on the previous line.

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 3]
```diff
@@ -111,12 +111,12 @@ class XMLProfileManager
...
```
Comment (MiguelCompany) [inline include/fastrtps/xmlparser/XMLProfileManager.h]: This should go on the previous line.

File: include/fastrtps/xmlparser/XMLProfileManager.h [hunk 4]
```diff
@@ -127,12 +127,12 @@ class XMLProfileManager
...
```
Comment (MiguelCompany) [inline include/fastrtps/xmlparser/XMLProfileManager.h]: This should go on the previous line.

File: include/fastrtps/xmlparser/XMLProfileManager.h (hunk 2 above)
Comment (EduPonz) [reply include/fastrtps/xmlparser/XMLProfileManager.h]: Done

File: include/fastrtps/xmlparser/XMLProfileManager.h (hunk 3 above)
Comment (EduPonz) [reply include/fastrtps/xmlparser/XMLProfileManager.h]: Done

File: include/fastrtps/xmlparser/XMLProfileManager.h (hunk 4 above)
Comment (EduPonz) [reply include/fastrtps/xmlparser/XMLProfileManager.h]: Done

**Comment by MiguelCompany** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/868/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/731/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/863/)

**Pull request merged**